tests/MongoDB.Driver.Core.Tests/Core/Connections/BinaryConnection_CommandEventTests.cs

```diff
@@ -38,6 +38,15 @@ namespace MongoDB.Driver.Core.Connections
 {
     public class BinaryConnection_CommandEventTests : IDisposable
     {
+        private static readonly IReadOnlyDictionary<string, Func<BsonDocument, bool>> __conditionallySecuritySensitiveCommands =
```

Comment:
Use reflector and nested class?

Reply:
It sounds like now we need only ShouldRedactMessage method?

Reply:
Hm... we could reflect on the private method `ShouldRedactMessage` this is true. However, I think we still need access to both the HashSet and Dictionaries in order to generate the names of the commands in `GetPotentiallyRedactedCommands`

Reply:
Low priority design comment:
I've used a reflector and an internal static class, but I'm not sure it was worth it: the variable names got so long I had to introduce an intermediate variable.

Reply:
This comment is out of date now since Dima has LGTMed and we've gone with Robert's low tech  approach of not reflecting against the data structures. Resolving.